packages/lit-dev-content/src/components/litdev-search.ts

```diff
@@ -41,26 +30,52 @@ interface UserFacingPageData {
   isSubsection: boolean;
 }
 
+/**
+ * Algolia result that returns stringified HTML that is highlighted.
+ */
+type HighlightOrSnippetResult<T> = {
+  [key in keyof T]: {
+    value: string;
+  };
+};
+
 /**
  * Subset of the suggestion returned by Algolia when there is a matching search
  * result.
  */
-type Suggestion = Omit<UserFacingPageData, 'text'>;
+type Suggestion = Omit<UserFacingPageData, 'text' | 'heading'> & {
+  _highlightResult: Pick<
+    HighlightOrSnippetResult<UserFacingPageData>,
+    'title' | 'heading'
+  >;
+  _snippetResult: Pick<HighlightOrSnippetResult<UserFacingPageData>, 'text'>;
+};
 
 /**
- * Returns a page suggestion TemplateResult with title and optional heading.
+ * Metadata for the type of search suggestion chip to display.
  */
-function titleAndHeadingCard(
-  title: string,
-  heading: string,
-  isSubsection: boolean
-): TemplateResult {
-  if (isSubsection) {
-    return html`<div class="title-and-header">
-      <span class="title">${title}</span><span class="header">${heading}</span>
-    </div>`;
-  }
-  return html`<span class="header">${title}</span>`;
+interface SuggestionType {
+  tag: string;
```

Comment:
Can this be made a stricter type (which can also be used in the `SuggestionGroup`.

`tag: 'other' | 'api' | 'docs' | 'article'`

Reply:
I plan to pull this out in an upcoming PR where I do this work on the indexer level. Can we punt for now?

Reply:
nvm just did it anyway